pkgs/applications/misc/moolticute/udev.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "moolticute-udev-${version}";
+  version = "0.30.8";
+
+  src = fetchurl {
+    url = "https://github.com/mooltipass/moolticute/archive/v${version}.tar.gz";
```

Comment:
Actually, I see this is same source as above. Why separate derivation then? Can't you install `$out/lib/udev/rules.d/50-mooltipass.rules` in moolticute `derivation`?

If it is really needed to have them separated, make separate output `udev` and install rules there.

Reply:
I want moolticute GUI app can be installed for user (`nix-env -i`) but the udev rules can't be per-user and they are installed into system (`/etc/nixos/`). That's why I've followed the approach for rfkill / rfkill_udev which have separate derivations. Is there another way to reach the goal? I guess separate output doesn't solve that.

Reply:
There is no problem to have udev rules installed into user Nix profile. What is the problem with that? They are not used, but don't harm also.

Reply:
Surely, unused udev rules in user Nix profile are ok. The problem I tried to solve is how to _install_ udev rules into the system if moolticute GUI app is in user Nix profile. My solution is to add `services.udev.packages = [ moolticute_udev ]` to `/etc/nixos/configuration.nix`. So the separate moolticute_udev doesn't depend on Qt and other GUI stuff not related with udev rules.

Reply:
ok then just install udev rules to separate output.
```
...
outputs = [ "out" "udev" ];
preInstall = ''
    mkdir -p $udev/lib/udev/rules.d
    sed -n '/^ \+cat > "$tmpfile" <<- EOF$/,/^EOF$/p' data/moolticute.sh | sed '1d;$d' \
      > $udev/lib/udev/rules.d/50-mooltipass.rules  
'';
...
```
Then you can set all-packages as:
```
  moolticute = libsForQt5.callPackage ../applications/misc/moolticute { };

  moolticute-udev = moolticute.udev; # or maybe just drop this?
```
Both `moolticute` and `moolticute.udev` should work as `services.udev.packages` packages. The second one won't depend on Qt, you can check that yourself with `nix-store -qR $(nix-build -A moolticute.udev)`

Reply:
I tried to split outputs. The problems is `nix-build -A moolticute.udev` fetches all the dependencies and builds binaries (although doesn't install them). That's why `udev.nix` doesn't specify libusb1, qmake, etc. and why it has a line `dontBuild = true`. Is it possible to specify different dependencies for multiple outputs? and to disable build step for one of them? I haven't found an answer in nixkpgs' manual.

Reply:
This is not a problem since the package will be fetched from binary cache most of the time and then only the dependencies required by the output will be needed.